api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -93,6 +93,21 @@ message VmConfig {
   // update and do a background fetch to fill the cache, otherwise fetch the code asynchronously and enter
   // warming state.
   bool nack_on_code_cache_miss = 6;
+
+  // Specifies environment variables to be injected to this VM which will be available through
+  // WASI's `environ_get` and `environ_get_sizes` system calls. Note that these functions are mostly implicitly
+  // called in your language's std libs, so you do not need to call them directly and you can access to env
+  // vars just like when you do on native platforms.
+  EnvironmentVariables environment_variables = 7;
+}
+
+message EnvironmentVariables {
+  // The keys of *Envoy's* environment variables exposed to this VM. In other words, if a key exists in Envoy's environment
+  // variables, then that key-value pair will be injected. Note that if a key does not exist, it will be ignored.
+  repeated string host_env_keys = 1;
```

Comment:
I wonder if we want some notion of env vars elsewhere in xDS as well, so we might want to factor this message to base types. @markdroth thoughts?